guides/plugins/plugins/framework/system-check/add-custom-check.md

```diff
@@ -0,0 +1,98 @@
+---
+nav:
+  title: Add custom check
+  position: 10
+
+---
+
+# Overview
+
+In this guide, we will be building a dummy example of a custom system check that verifies if the local system has enough disk space to operate normally.
+
+## Add a new Custom Check
+
+First, you need to add a new `LocalDiskSpaceCheck` class that extends the `Shopware\Core\Framework\SystemCheck\BaseCheck` and implement the essential categorization methods.
+
+### Fill the categorization methods
+
+Each check contains a set of categorization methods that help to classify the check, and determine when and where it should be executed.
+
+```php
+class LocalDiskSpaceCheck extends BaseCheck
+{
+    public function category(): Category
+    {
+        // crucial for the system to function at all. 
+        return Category::SYSTEM;
+    }
+
+    public function name(): string
+    {
+        return 'LocalDiskSpaceCheck';
+    }
+
+    protected function allowedSystemCheckExecutionContexts(): array
+    {   // a potentially long-running check, because it has an IO operation.
+        return SystemCheckExecutionContext::longRunning();
+    }
+}
+```
+
+### Create the check logic
+
+The next step is to implement the actual check logic. We will check if the disk space is below a certain threshold and return the appropriate result.
+
+```php
+class LocalDiskSpaceCheck extends BaseCheck
+{
+    public function __construct(
+        private readonly string $adapterType,
+        private readonly string $installationPath,
+        private readonly int $warningThresholdInMb
+    )
+    {
+    }
+
+    public function run(): Result
+    {
+        if ($this->adapterType !== 'local') {
+           return new Result(name: $this->name(), status: Status::SKIPPED, message: 'Disk space check is only available for local file systems.', healthy: true)
```

Comment:
Maybe a little description about the healthy flag might be useful, because it could be confusing when a result can have a status of error but also be healthy?